app/models/katello/concerns/http_proxy_extensions.rb

```diff
@@ -17,16 +20,33 @@ def update_default_proxy_setting
         changes = self.previous_changes
         if changes.key?(:name)
           previous_name = changes[:name].first
-          setting = Setting::Content.unscoped.
-            where(name: "content_default_http_proxy").
-            first
+          setting = Setting.find_by(name: 'content_default_http_proxy')
 
           if setting && setting.value == previous_name && !previous_name.blank?
             setting.update_attribute(:value, self.name)
           end
         end
       end
 
+      def update_repository_proxy_details
+        changes = self.previous_changes
+        if changes.key?(:url) || changes.key?(:username) || changes.key?(:password)
+
+          root_repos = RootRepository.with_selected_proxy(id)
+
+          if self == HttpProxy.default_global_content_proxy
+            root_repos += RootRepository.with_global_proxy
+          end
+
+          repos = root_repos.collect(&:repositories)
```

Comment:
you probably could just collect  (&:library_instance), we don't to update the cv repos